evolver/calibration/actions.py

```diff
@@ -86,13 +105,34 @@ def execute(self, state: Dict[str, Any], payload: UserInput) -> Dict[str, Any]:
         if not reference_values or not raw_values:
             raise ValueError(f"Insufficient data to calculate fit for {hardware_name} {vial_key}")
```

Comment:
I'm not sure if this is a good idea, see my other comment above/below about data correctness being the responsibility of the transformer.

Reply:
yep, refactored to let Transformer be the source of these errors and let them bubble up.